modules/activitypub/keypair.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package activitypub
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+)
+
+const rsaBits = 2048
+
+// GenerateKeyPair generates a public and private keypair for signing actions by users for activitypub purposes
+func GenerateKeyPair() (string, string, error) {
+	priv, _ := rsa.GenerateKey(rand.Reader, rsaBits)
+	privPem, err := pemBlockForPriv(priv)
+	if err != nil {
+		return "", "", err
+	}
+	pubPem, err := pemBlockForPub(publicKey(priv))
```

Comment:
I don't think the `publicKey` function is needed as it is called only here.
Additionally, the function declaration seems overly complicated to me.
If I understand go syntax correctly, the following should still be valid, right?
```suggestion
	pubPem, err := pemBlockForPub(&priv.PublicKey)
```